data/src/main/java/org/apache/iceberg/data/PartitionStatsHandler.java

```diff
@@ -150,6 +161,75 @@ public static PartitionStatisticsFile computeAndWriteStatsFile(Table table, long
         table, snapshot.snapshotId(), schema(partitionType), sortedStats);
   }
 
+  /**
+   * Incrementally computes the stats after the snapshot that has partition stats file till the
+   * given snapshot and writes the combined result into a {@link PartitionStatisticsFile} after
+   * merging the stats.
+   *
+   * @param table The {@link Table} for which the partition statistics is computed.
+   * @param snapshotId snapshot for which partition statistics are computed.
+   * @return {@link PartitionStatisticsFile} for the given snapshot, or null if no statistics are
+   *     present.
+   */
+  public static PartitionStatisticsFile computeAndWriteStatsFileIncremental(
+      Table table, long snapshotId) throws IOException {
+    Preconditions.checkArgument(table != null, "Table cannot be null");
+    Snapshot snapshot = table.snapshot(snapshotId);
+    Preconditions.checkArgument(snapshot != null, "Snapshot not found: %s", snapshotId);
+
+    StructType partitionType = Partitioning.partitionType(table);
+    Schema statsFileSchema = schema(partitionType);
+    PartitionStatisticsFile statisticsFile = latestStatsFile(table, snapshotId);
+    Collection<PartitionStats> stats;
+    if (statisticsFile == null) {
+      LOG.info("Previous stats not found. Computing the stats for whole table.");
+      stats = PartitionStatsUtil.computeStats(table, table.snapshot(snapshotId));
+    } else {
+      PartitionMap<PartitionStats> statsMap = PartitionMap.create(table.specs());
+      // read previous stats
+      try (CloseableIterable<PartitionStats> oldStats =
+          readPartitionStatsFile(statsFileSchema, Files.localInput(statisticsFile.path()))) {
```

Comment:
since the new unified tuple is used for reading the old stats file. It automatically handled the schema evolution.